plugins/modules/net_tools/dnsimple_info.py

```diff
@@ -0,0 +1,337 @@
+
+
+def build_url(account, key, is_sandbox):
+    headers = {'Accept': 'application/json',
+               'Authorization': 'Bearer ' + key}
+    url = 'https://api{sandbox}.dnsimple.com/'.format(
+        sandbox=".sandbox" if is_sandbox else "") + 'v2/' + account
+    req = Request(url=url, headers=headers)
+    prepped_request = req.prepare()
+    return prepped_request
+
+
+def iterate_data(module, request_object):
+    base_url = request_object.url
+    response = Session().send(request_object)
+    if 'pagination' in response.json():
+        data = response.json()["data"]
+        pages = response.json()["pagination"]["total_pages"]
+        if int(pages) > 1:
+            for page in range(1, pages):
+                page = page + 1
+                request_object.url = base_url + '&page=' + str(page)
+                new_results = Session().send(request_object)
+                data = data + new_results.json()["data"]
+        return(data)
+    else:
+        module.fail_json('API Call failed, check ID, key and sandbox values')
+
+
+def record_info(dnsimple_mod, req_obj):
+    req_obj.url, req_obj.method = req_obj.url + '/zones/' + dnsimple_mod.params["name"] + '/records?name=' + dnsimple_mod.params["record"], 'GET'
+    return iterate_data(dnsimple_mod, req_obj)
+
+
+def domain_info(dnsimple_mod, req_obj):
+    req_obj.url, req_obj.method = req_obj.url + '/zones/' + dnsimple_mod.params["name"] + '/records?per_page=100', 'GET'
+    return iterate_data(dnsimple_mod, req_obj)
+
+
+def account_info(dnsimple_mod, req_obj):
+    req_obj.url, req_obj.method = req_obj.url + '/zones/?per_page=100', 'GET'
+    return iterate_data(dnsimple_mod, req_obj)
+
+
+def main():
+    # define available arguments/parameters a user can pass to the module
+    fields = {
+        "account_id": {"required": True, "type": "str"},
+        "api_key": {"required": True, "type": "str", "no_log": True},
+        "name": {"required": False, "type": "str"},
+        "record": {"required": False, "type": "str"},
+        "sandbox": {"required": False, "type": "bool", "default": False}
+    }
+
+    result = {
+        'changed': False
+    }
+
+    module = AnsibleModule(
+        argument_spec=fields,
+        supports_check_mode=True
+    )
+
+    params = module.params
+    req = build_url(params['account_id'],
+                    params['api_key'],
+                    params['sandbox'])
+
+    if not HAS_ANOTHER_LIBRARY:
+        # Needs: from ansible.module_utils.basic import missing_required_lib
+        module.exit_json(
+            msg=missing_required_lib('another_library'),
+            exception=ANOTHER_LIBRARY_IMPORT_ERROR)
+
+    # At minimum we need account and key
+    if params['account_id'] and params['api_key']:
+        # If we have a record return info on that record
+        if params['name'] and params['record']:
+            result['dnsimple_record_info'] = record_info(module, req)
+            module.exit_json(changed=False, msg=result)
```

Comment:
You already include `changed`, and if you package everything in `msg` you break both the users expectations by not having the top-level return values you documented, and by not having `msg` a string.
```suggestion
            module.exit_json(**result)
```